File: src/api/apis/binance.js
const { API } = require('./api.js')

class BinanceAPI extends API {
  // debug = true

  _request(method = 'GET', pathname = '/', { query = {} } = {}) {
    return fetch(`https://api4.binance.com${pathname}?${(new URLSearchParams(query)).toString()}`, { method }).then((res) => res.json())
  }

  klines(symbol = 'BNBBRL', interval = '1s', limit = 10) {
    this._request('GET', '/api/v3/klines', { query: { symbol, interval, limit } })
      .then((json) => this._response('klines', json))
      .catch((err) => this._error(err))
  }
}

module.exports = { BinanceAPI }

File: src/cli/bin/binance.dart
import 'package:http/http.dart';

void main() async {
  var response = await get(Uri.https(
    'api4.binance.com',
    '/api/v3/klines',
    { 'symbol': 'BTCBRL', 'interval': '1m', 'limit': '10' }
  ));

  print(response.body);
}
